spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -355,4 +355,21 @@ public boolean reportColumnStats() {
         .defaultValue(SparkSQLProperties.REPORT_COLUMN_STATS_DEFAULT)
         .parse();
   }
+
+  public boolean deriveStatsFromManifestSessionConf() {
+    return confParser
+        .booleanConf()
+        .sessionConf(SparkSQLProperties.DERIVE_STATS_FROM_MANIFEST_ENABLED)
+        .defaultValue(SparkSQLProperties.DERIVE_STATS_FROM_MANIFEST_ENABLED_DEFAULT)
+        .parse();
+  }
+
+  public boolean deriveStatsFromManifestTableProperty() {
+    return confParser
+        .booleanConf()
+        .option(SparkReadOptions.DERIVE_STATS_FROM_MANIFEST_ENABLED)
+        .tableProperty(TableProperties.DERIVE_STATS_FROM_MANIFEST_ENABLED)
+        .defaultValue(TableProperties.DERIVE_STATS_FROM_MANIFEST_ENABLED_DEFAULT)
+        .parse();
+  }
```

Comment:
This table-level property takes precedence over the session configuration when it is turned off, enabling users to derive statistics only for a specific table.